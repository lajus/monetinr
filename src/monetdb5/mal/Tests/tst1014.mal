# now within a function to assure that reference counting
# takes care of releasing the variables.

function test();
    io.print("test");
    bbp.open();
    x:= bbp.getObjects();
    io.print(x);
    io.print("test");
    b:= bat.new(:oid,:int);
    bat.append(b,2);
    bbp.deposit("myfirstbat",b);
    c:= bbp.take("myfirstbat");
    io.print(c);
    bbp.close();
# at the end of the call, the BBP should have discarded
# the temporary bats of the user
end test;

user.test();
io.print("continue");
bbp.open();
x:= bbp.getObjects();
xr:= bat.reverse(x);
xp:= algebra.project(xr);
io.print(xp);
v:= bbp.take("myfirstbat");
io.print(v);
bbp.close();
