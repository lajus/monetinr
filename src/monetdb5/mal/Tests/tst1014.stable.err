stderr of test 'tst1014` in directory 'monetdb5/mal` itself:
