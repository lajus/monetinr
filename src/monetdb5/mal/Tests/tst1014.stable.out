stdout of test 'tst1014` in directory 'monetdb5/mal` itself:


# 07:58:54 >  
# 07:58:54 >  Mtimeout -timeout 60 Mserver "--config=/ufs/mk/monet5/Linux/etc/MonetDB5.conf" --debug=10 --set "monet_mod_path=/ufs/mk/monet5/Linux/lib/MonetDB5:/ufs/mk/opensource/MonetDB/Linux/lib/MonetDB" --set "gdk_dbfarm=/ufs/mk/monet5/Linux/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/mk/monet5/Linux/var/MonetDB5/log" --set mapi_port=42158 --set sql_port=55708 --set monet_prompt= --trace "--config=/ufs/mk/monet5/src/mal/Tests/All.conf" --dbname=mTests_src_mal  tst1014.mal </dev/null
# 07:58:54 >  

# Monet Database Server V4.99.19
# Copyright (c) 1993-2004, CWI. All rights reserved.
# Compiled for i686-redhat-linux-gnu/32bit; dynamically linked.
# config:/ufs/mk/monet5/Linux/etc/MonetDB5.conf
# dbfarm:/ufs/mk/monet5/Linux/var/MonetDB5/dbfarm
# dbname:mTests_src_mal
# Visit http://monetdb.cwi.nl/ for further information.
## now within a function to assure that reference counting
## takes care of releasing the variables.
#
#function test();
#    print("test");
#    bbp.open();
#    x:= bbp.getObjects();
#    print(x);
#    print("test");
#    b:= new(:int,:int);
#    insert(b,1,2);
#    bbp.deposit("myfirstbat",b);
#    c:= bbp.take("myfirstbat");
#    print(c);
#    bbp.close();
## at the end of the call, the BBP should have discarded
## the temporary bats of the user
#end test;
function user.test():void;
    io.print("test");
    bbp.open();
    x := bbp.getObjects();
    io.print(x);
    io.print("test");
    b := bat.new(:oid,:int);
    bat.append(b,2);
    bbp.deposit("myfirstbat",b);
    c := bbp.take("myfirstbat");
    io.print(c);
    bbp.close();
# at the end of the call, the BBP should have discarded 
# the temporary bats of the user 
end test;
#
#test();
#print("continue");
#bbp.open();
#x:= bbp.getObjects();
#print(x);
#v:= bbp.take("myfirstbat");
#print(v);
#bbp.close();
#
function user.main():void;
# now within a function to assure that reference counting 
# takes care of releasing the variables. 
    user.test();
    io.print("continue");
    bbp.open();
    x := bbp.getObjects();
    xr := bat.reverse(x);
    xp := algebra.project(xr);
    io.print(xp);
    v := bbp.take("myfirstbat");
    io.print(v);
    bbp.close();
end main;
[ "test" ]
#-------------------------#
# bbp	objects		  # name
# int	str		  # type
#-------------------------#
[ "test" ]
#-----------------#
# h	tmp_5	  # name
# int	int	  # type
#-----------------#
[ 0@0,	  2	  ]
[ "continue" ]
#-------------------------#
# bbp	objects		  # name
# int	str		  # type
#-------------------------#
[ "myfirstbat",	  nil	  ]
#-----------------#
# h	tmp_5	  # name
# int	int	  # type
#-----------------#
[ 0@0,	  2	  ]

# 07:58:55 >  
# 07:58:55 >  Done.
# 07:58:55 >  

